Testscripts/Windows/PRODUCTION-CHECKPOINT-PARTITION.ps1

```diff
@@ -135,27 +135,68 @@ function Main {
+                    $disk_combination += $suffix
+                }
+                $prefix = $rest[0]
+                $rest.RemoveAt(0)
+            })
+            $disk_combination += $disk_combination[-1..($disk_combination.Length-2)]
+        } else {
+            $disk_combination = $availableDisks
+        }
+
+        $testResultTmp = $resultFail
+        for ($i = 0; $i -le $disk_combination.Length; $i++) {
+            $mount_1st_disk = $disk_combination[$i]
             Run-LinuxCmd -username $user -password $password -ip $Ipv4 -port $VMPort `
-                -command "mount /dev/${mount_2nd_disk}1 /mnt/1; mount /dev/${mount_2nd_disk}2 /mnt/2" -runAsSudo
+                -command "mount /dev/${mount_1st_disk}1 /mnt/1; mount /dev/${mount_1st_disk}2 /mnt/2" -runAsSudo
+            if ($TestParams.DUALMOUNT) {
+                $i++
+                $mount_2nd_disk = $disk_combination[$i]
+                Run-LinuxCmd -username $user -password $password -ip $Ipv4 -port $VMPort `
+                    -command "mount /dev/${mount_2nd_disk}1 /mnt/1; mount /dev/${mount_2nd_disk}2 /mnt/2" -runAsSudo
+            }
+            $sts1 = Check-FileInLinuxGuest -VMPassword $password -VMPort $VMPort -VMUserName $user -Ipv4 $Ipv4 -fileName "/mnt/1/${testfile1}"
+            $sts2 = Check-FileInLinuxGuest -VMPassword $password -VMPort $VMPort -VMUserName $user -Ipv4 $Ipv4 -fileName "/mnt/2/${testfile1}"
+            if (-not $sts1 -or -not $sts2)
+            {
+                Write-LogWarn "TestFile1 is not present, try other disk"
+                if ($TestParams.DUALMOUNT) {
+                    Run-LinuxCmd -username $user -password $password -ip $Ipv4 -port $VMPort `
+                        -command "umount /dev/${mount_2nd_disk}1; umount /dev/${mount_2nd_disk}2" -runAsSudo
+                }
+                Run-LinuxCmd -username $user -password $password -ip $Ipv4 -port $VMPort `
+                    -command "umount /dev/${mount_1st_disk}1; umount /dev/${mount_1st_disk}2" -runAsSudo
+                continue
+            } else {
+                $testResultTmp = $resultPass
+                break
+            }
         }
-        $sts1 = Check-FileInLinuxGuest -VMPassword $password -VMPort $VMPort -VMUserName $user -Ipv4 $Ipv4 -fileName "/mnt/1/${testfile1}"
-        $sts2 = Check-FileInLinuxGuest -VMPassword $password -VMPort $VMPort -VMUserName $user -Ipv4 $Ipv4 -fileName "/mnt/2/${testfile1}"
-        if (-not $sts1 -or -not $sts2)
-        {
+
+        if( $testResultTmp -eq $resultPass){
+            $sts1 = Check-FileInLinuxGuest -VMPassword $password -VMPort $VMPort -VMUserName $user -Ipv4 $Ipv4 -fileName "/mnt/1/${testfile2}"
+            $sts2 = Check-FileInLinuxGuest -VMPassword $password -VMPort $VMPort -VMUserName $user -Ipv4 $Ipv4 -fileName "/mnt/2/${testfile2}"
+            if ($sts1 -or $sts2) {
+                Write-LogErr "TestFile2 is present,it should not be present on the VM"
```

Comment:
What does it mean? I don't understand what logic ends up here inside of the if-block statement.

Reply:
**TestFile2** is created after the Checkpoint-VM, so it should not be present after Restore-VMSnapshot.
Similarly, **TestFile1** is created before the Checkpoint-VM, so it should be present after Restore-VMSnapshot.